src/core/util/uri.cc

```diff
@@ -344,6 +362,23 @@ struct QueryParameterFormatter {
   }
 };
 
+struct Authority {
+  std::string user_info;
```

Comment:
These fields can be `absl::string_view` instead of `std::string`, to avoid unnecessary allocations and string copies.